pkg/monitoring/monitoring.go

```diff
@@ -0,0 +1,285 @@
+// Copyright 2021 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Package monitoring provides functionality to report metrics to Google Cloud Monitoring a.k.a.
+// Stackdriver.
+package monitoring
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	monitoring "cloud.google.com/go/monitoring/apiv3"
+	"github.com/golang/protobuf/ptypes/timestamp"
+	gax "github.com/googleapis/gax-go/v2"
+	"google.golang.org/genproto/googleapis/api/label"
+	"google.golang.org/genproto/googleapis/api/metric"
+	"google.golang.org/genproto/googleapis/api/monitoredres"
+	monitoringpb "google.golang.org/genproto/googleapis/monitoring/v3"
+)
+
+const (
+	// Metric types for metrics reported by this package to Cloud Monitoring. See docs for "type" in
+	// https://cloud.google.com/monitoring/api/ref_v3/rest/v3/projects.metricDescriptors#resource:-metricdescriptor.
+
+	// BEGIN Metrics for Toolchain Configs Generation
+	//
+	// All toolchain config generation metrics have the following characterics:
+	// 1. Cummulative integer tracking the number of times a toolchain configs release step
+	//    (generation, upload & test) runs to completion.
+	// 2. Each metric includes the following labels:
+	//    a. docker_image- A string representing the OS name of the toolchain docker image. e.g.,
+	//                     "rbe-ubuntu1604".
+	//    b. result- Bool set to true if the step succeeded.
+	//
+	// mtypeToolchainConfigsGenRuns tracks successful runs of rbe_configs_gen i.e., configs
+	// generation.
+	mtypeToolchainConfigsGenRuns = "custom.googleapis.com/toolchain_configs/generation/runs"
```

Comment:
```suggestion
	mtypeToolchainConfigsGenRuns = "custom.googleapis.com/rbe_toolchain_configs/generation/runs"
```

RBE is a relatively unique global namespace term, compared to "toolchain".  Ditto for the other metrics.

Reply:
Done. Replaced with "rbe/bazel-toolchains" because these configs are specific to Bazel as well